back/src/database/SpotSQL.js

```diff
@@ -19,40 +21,85 @@ module.exports.saveSpot = function(newSpot){
         text: 'DELETE from spots.spots where spot_id = $1;',
         values: [newSpot.spotId]
     };
-    return connection.connect().then(()=>{
-        //TODO: NULL CHECK
-        return connection.query(query1)
-        .then(()=>{
-            return connection.query(query2)
-            .then(()=>{
-                connection.end()
-                debug(fileLabel,"saved spot: " + newSpot);
-                return {"success":true,"data":newSpot};
-            })
-            .catch((exception)=>{
-                error(fileLabel,"Error while saving review: " + exception);
-                return connection.query(deleteSpotQuery)
-                .then(()=>{
-                    connection.end()
-                    debug(fileLabel,"deleted spot: " + newSpot);
-                    return {"success":false,"data":exception};
-                })
-                .catch(()=>{
-                    error(fileLabel,"Error while deleting spot: " + exception);
-                    return {"success":false,"data":exception};
-                })
-            });
+    const client = await pool.connect();
+    //TODO: NULL CHECK
+    if((isEmpty(newSpot.spotName))){
+        error(fileLabel,"Spot Name is Empty!!!!")
+        return {"success":false,"data":"Spot Name is Empty!!!!"};
+    }
+    return client.query(query1).then(()=>{
+        return client.query(query2).then(()=>{
+            client.release();
+            info(fileLabel,"saved spot: " + newSpot);
+            return {"success":true,"data":newSpot};
         })
-        .catch((exception)=>{
-            connection.end()
-            error(fileLabel,"Error while saving spot: " + exception);
-            return {"success":false,"data":exception};
+        .catch(err=>{
+            client.release();
```

Comment:
if you client.release() here, you won't be able to execute deleteSpotQuery after